src/util/magic.ts

```
@@ -11,7 +12,28 @@ export async function getMagic(): Promise<Magic> {
   return magicInstance;
 }
 
-export async function getMagicUserMetadataById(userId: string): Promise<MagicUserMetadata> {
+export async function getMagicUserMetadataByDIDToken(didToken: string): Promise<MagicUserMetadata> {
   const magic = await getMagic();
-  return magic.users.getMetadataByIssuer(userId);
+  return magic.users.getMetadataByToken(didToken);
+}
+
+export function verifyEmailByMagicUserMetadata(
+  email: string,
+  magicUserMetadata: MagicUserMetadata,
+): boolean {
+  if (!magicUserMetadata.email) {
+    return false;
+  }
+  if (email !== magicUserMetadata.email) {
+    throw new ValidationError('MAGIC_EMAIL_MISMATCH');
```

Not sure if it is a good idea to throw instead of just return false here, seems one would expect this to return false judging by function name